Add bound method tests for async_execution with RRef helper

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#44716 Add bound method tests for async_execution with RRef helper**
* #44666 Make async_execution compatible with RRef helpers
* #44663 Add _get_type() API to RRef

Differential Revision: [D23707326](https://our.internmc.facebook.com/intern/diff/D23707326)

mrshenli committed:
Add bound method tests for async_execution with RRef helper

[ghstack-poisoned]

mrshenli committed:
Update on "Add bound method tests for async_execution with RRef helper"




[ghstack-poisoned]

rohan-varma reviewed (approved):
Does "bound" here mean that the futures are chained?

mrshenli commented:
> Does "bound" here mean that the futures are chained?

I was referring to Python's terminology:

```
>>> class Test:
...   def f(self):
...     pass
... 
>>> t = Test()
>>> t.f
<bound method Test.f of <__main__.Test object at 0x7f06a08f8610>>
```

mrshenli committed:
Update on "Add bound method tests for async_execution with RRef helper"


Differential Revision: [D23707326](https://our.internmc.facebook.com/intern/diff/D23707326)

[ghstack-poisoned]

mrshenli committed:
Update on "Add bound method tests for async_execution with RRef helper"


Differential Revision: [D23707326](https://our.internmc.facebook.com/intern/diff/D23707326)

[ghstack-poisoned]
